base/associative.jl

```diff
@@ -451,6 +451,16 @@ function rehash!(t::ObjectIdDict, newsz = length(t.ht))
     t
 end
 
+function sizehint!(t::ObjectIdDict, newsz)
+    newsz = _tablesz(newsz*2)  # *2 for keys and values in same array
+    oldsz = length(t.ht)
+    # grow at least 25%
+    if newsz < (oldsz*5)>>2
+        return t
```

Comment:
Yes, most definitions of `sizehint!` (e.g. Array and Dict) will choose to do nothing in some cases. Here you want to avoid the cost of rehashing.